### DISCUSSION TITLE
No module named OCP

### DISCUSSION THREAD

**Discussion by hootsh1337** [CadQuery/cadquery#1426] [Q&A] [answered]
Hello, I tried installing cadquery using pip install. But the following error happens when I run: import cadquery 

>   File "C:\Users\H\AppData\Local\Programs\Python\Python312\Lib\site-packages\cadquery\__init__.py", line 10, in <module>
>     from .occ_impl.geom import Plane, BoundBox, Vector, Matrix, Location
>   File "C:\Users\H\AppData\Local\Programs\Python\Python312\Lib\site-packages\cadquery\occ_impl\geom.py", line 5, in <module>
>     from OCP.gp import (
> ModuleNotFoundError: No module named 'OCP'

I tried pip install OCP but it didn't work.

Both Python and pip are latest versions. Am I doing anything wrong?

I also tried searching the forums but none of the answers I found work for me.

Thank you very much and apologies in advance if this is a double post, I've honestly been searching for a solution for hours.

**Comment by adam-urbanczyk** [comment, marked as answer]
OCP is not yet built for Python 3.12 AFAIR.

**Reply by hootsh1337** [reply]
Ah, that makes sense. Thank you. 👍
What version should I install?

**Reply by jmwright** [reply]
Python 3.10 is the safest. I've seen a few edge cases with 3.11.

**Reply by hootsh1337** [reply]
OK will use that, thank you.

**Reply by ctrl-Sebastian** [reply]
What did you do after? i have 3.10 but got the same error

![Uploading image.png…]()

**Reply by jmwright** [reply]
@ctrl-Sebastian The image you posted shows as broken for me. How did you install CadQuery? If you did a pip install in a conda environment the dependences will sometimes not install correctly and you will need to install them manually.

**Reply by hootsh1337** [reply]
@ctrl-Sebastian What I did after, if I remember correctly, is simply using 3.10, solved the problem.

I can't see your image; the link is broken somehow. But there are some OS differences, though. From my limited experience working with it. I ran into a similar issue a while ago trying to use CadQuery with Django on a server but due to OS issues kept also showing us similar errors with OCP...etc. So, it worked with Python 3.10 on my Windows 10 machine, but not on the server cuz different os.

**Reply by ctrl-Sebastian** [reply]
![image](https://github.com/user-attachments/assets/2d23d023-161f-4776-a6c3-88ade7ba8797)
it was this image, 
im not using a conda environment, just a virtualenv with python 3.10, after adding OCP to hidden imports i got this new error:
![image](https://github.com/user-attachments/assets/300f9205-de0a-4687-a5f6-0006fa519b3f)

so then i added casadi._casadi to the hidden imports just in case and got:

![image](https://github.com/user-attachments/assets/6a5e8ce2-809f-4ec1-a595-9449d9b9ba00)

after running 
`pip install git+https://github.com/CadQuery/cadquery.git`
i got:

![image](https://github.com/user-attachments/assets/8f1f7bbe-fbfb-432f-9a91-dc0ecbc1f7aa)

**Reply by ctrl-Sebastian** [reply]
if i try pip install OCP or casadi or vtk it just says that the requirement is met

**Comment by nexovec** [comment]
Also getting this error on python 3.13.

**Reply by ctrl-Sebastian** [reply]
I solved it, 

As @adam-urbanczyk said OCP is not yet built for Python 3.12, I switched to 3.10 (specifically 3.10.11) and used pyinstaller but with the .spec file, whenever i tried to make the executable i got an import module error that was solved by adding the specified module into the hidden imports array of the .spec file, each time i tried making the executable, another module was missing, that was until casadi was missing, for that i had to add it to the import array and also, the binaries weren't found so i had to also add the binaries location to the .spec file.

The example:

a = Analysis(
    binaries=[(r'C:/PATH_TO_PROJECT/.venv/lib/site-packages/casadi/', 'casadi')],
    hiddenimports=['OCP', 'casadi._casadi', '_casadi'],
)

**Reply by ctrl-Sebastian** [reply]
Also, i got an error message for the icon of my project saying that it was missing, i suppose you can add it int he spec file but i just dragged it into the dist/_internal directory. Hope it helps.

**Reply by nexovec** [reply]
It says that reply is two years old, so I assumed that has changed, I stand corrected. Maybe python isn't for me. :smile: I will try to use something else, but thanks for help.
